src/python/pants/backend/python/macros/pex_binaries_from_sources_caof.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os.path
+
+from typing import Iterable
+
+from pants.engine.target import InvalidFieldException
+from pants.backend.python.macros.caof_utils import (
+    OVERRIDES_TYPE,
+)
+
+class PexBinariesFromSourcesCAOF:
+    """Translates N sources to equivalent `pex_binary` targets with entry_point set to the source."""
+
+    def __init__(self, parse_context):
+        self._parse_context = parse_context
+
+    def __call__(
+        self,
+        *,
+        sources: str = Iterable[str],
+        overrides: OVERRIDES_TYPE = None,
+    ) -> None:
+        overrides = overrides.copy() or {}
```

Comment:
Could implement this without `copy` and `pop`, but makes the code easier to follow IMO. Let me know if I should re-write in the other way.